feat(docs): Add documentation on upgrading from Node 6 (#14785)
gatsbyjs/gatsby: docs/docs/upgrading-node-js.md

> # Upgrading your Node.js version

Reviewer: ```suggestion
---
title: Upgrading your Node.js version
---
```

> ## Introduction

Reviewer: Probably just remove?

```suggestion
```

## Gatsby's Node.js support policy

Gatsby aims to support any version of Node that has a release status of _Current_, _Active_, or _Maintenance_. Once a major version of Node reaches _End of Life_ status Gatsby will stop supporting that version.

Gatsby will stop supporting the _End of Life_ Node release in a minor version.

Check [Node's releases document](https://github.com/nodejs/Release#nodejs-release-working-group) for version statuses.

## What version of Node do I have?

Run `node -v` in a terminal to see which version of Node you have.

```
node -v
v10.16.0
```

> This example shows Node version 10.

Reviewer: Un-opinionated on this 🤷‍♂ could be worth doing. Up to you!

```suggestion
This example shows Node version 10, specifically v10.16.0.
```

Author: It's a good one 👍

## Upgrading from Node version 6

> Node version 6 reached _End-of-life_ status on 30th April 2019. Many of Gatsby's dependencies are updating to Node version 8 and above. Gatsby must also update in order release new features and bug fixes.

Reviewer: ```suggestion
Node version 6 reached _End-of-life_ status on 30th April 2019. Many of Gatsby's dependencies are updating to Node version 8 and above. Gatsby must also update in order to deliver new features and bug fixes more quickly.
```

> Generally we'd recommend using [the Node version who's status is _Active LTS_](https://github.com/nodejs/Release#nodejs-release-working-group) (Node 10 at time of writing). However, in this document you'll learn how to update from Node 6 to Node 8 as this is likely to be the least disruptive upgrade for you.

Reviewer: ```suggestion
Generally we'd recommend using [the Node version whose status is _Active LTS_](https://github.com/nodejs/Release#nodejs-release-working-group) (Node 10 at time of writing). However, in this document you'll learn how to update from Node 6 to Node 8 as this is likely to be the least disruptive upgrade for you.
```

Author: I thought that looked wrong :)

> What about Node 7? Stable versions of Node are evenly numbered releases - Node 6, Node 8, Node 10 etc. Only use uneven release numbers if you'd like to try cutting-edge and experimental features.

There are multiple ways to update your version of Node depending on how you originally installed it. Read on to find the best approach for you.

### Using Homebrew

> This is our recommended way to install a newer version of Node.

Reviewer 2: This applies to OSX only, but it's not mentioned anywhere

Author: It's on Linux and Windows (subsystem for Linux) too! See https://www.gatsbyjs.org/tutorial/part-zero/#install-homebrew-for-nodejs

Reviewer 2: Oh, TIL

==> Formulae
heroku/brew/heroku-node ✔        llnode                           node@10                          nodebrew
leafnode                         node ✔                           node@8                           nodeenv
libbitcoin-node                  node-build                       node_exporter                    nodenv
```

You're interested in the next stable version of Node after Node 6, which is Node 8. Homebrew makes this available in a package called `node@8`. Run:

```
brew install node@8
```

Once that's complete, run:

```
node -v
```

to confirm that you've upgraded from Node version 6 up to version 8.

### Using a Node version management package

There are two popular packages used for managing multiple versions of Node on your system. Use one of these to update to a newer version of Node if they're already available on your computer.

These packages are very useful for people that regularly work with different versions of Node.

#### n

Run:

```
n
```

in a terminal to see if n is installed on your system. If it's installed, you can run `n 8` to install and use Node version 8.

[Check n's documentation for further instructions](https://github.com/tj/n).

> #### nvm

Reviewer: Nit: but I'd maybe move nvm above n? Perhaps it's just my bubble--but it seems that nvm is much more heavily used.

Author: Makes sense, will update 👍 

I use `nvm` but have to refer to the docs _every_ time I use it. `n`'s API is much simpler in comparison.

Run

```
nvm
```

in a terminal to see if nvm is installed on your system. If it's installed, you can run:

```
nvm install 8
nvm alias default 8
```

to install and use Node version 8.

[Check nvm's documentation for further instructions](https://github.com/nvm-sh/nvm).

### Installing from nodejs.org

If you aren't using any of the previously listed installation methods, you can [download a Node installer directly from nodejs.org](https://nodejs.org/en/).

> Gatsby's recommended way to install Node is by using Homebrew. See the Homebrew section of this document for more info.

Reviewer: ```suggestion
Gatsby's recommended way to install Node is by using Homebrew. Refer to the previous [Homebrew section of this document](#using-homebrew) for more info.
```

## Conclusion

Gatsby takes backwards compatibility seriously and aims to support older versions of Node for as long as possible. We understand that juggling different software versions is not a productive way to spend your day.

Gatsby also relies on a huge ecosystem of JavaScript dependencies. As the ecosystem moves away from older, unsupported Node versions we have to keep pace to ensure that bugs can be fixed and new features can be released.

> In this document you learned how you upgrade from Node version 6 (which has reached _End of Life_ status), to Node version 8 (which is reached _Maintenance_) status.

Reviewer: ```suggestion
In this document you learned how you upgrade from Node version 6 (which has reached _End of Life_ status), to Node version 8 (which has reached _Maintenance_) status.
```

Author: This has a good correction, thanks :)
